src/hotspot/share/jvmci/jvmciCompilerToVMInit.cpp

```diff
@@ -227,12 +236,21 @@ void CompilerToVM::Data::initialize(JVMCI_TRAPS) {
     assert(base != nullptr, "unexpected byte_map_base");
     cardtable_start_address = base;
     cardtable_shift = CardTable::card_shift();
+  } else if (bs->is_a(BarrierSet::ShenandoahBarrierSet)) {
+    cardtable_shift = CardTable::card_shift();
```

Comment:
I understand the barrier code does not use `cardtable_start_address`, but should we still initialize it here to `nullptr`?